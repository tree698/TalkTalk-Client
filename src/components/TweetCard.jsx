import React from 'react';
import { formatAgo } from '../util/date';
import Avatar from './Avatar';

export default function TweetCard({ owner, tweet }) {
  const { id, text, createdAt, userId, username, photo } = tweet;

  return (
    <li>
      <div>
        <Avatar photo={photo} username={username} />
      </div>
      <div>
        <p>{username}</p>
        <p>{formatAgo(createdAt)}</p>
      </div>
      <p>{text}</p>
    </li>
  );
}
