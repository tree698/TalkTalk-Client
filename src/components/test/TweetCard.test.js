import TweetCard from '../TweetCard';
import renderer from 'react-test-renderer';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('TweetCard', () => {
  let fakeTweet;
  beforeEach(() => {
    fakeTweet = {
      id: 'test',
      text: 'test',
      createdAt: new Date(),
      username: 'test',
      photo: 'test',
    };
  });

  it('renders with owner correctly', () => {
    const component = renderer.create(
      <TweetCard owner="fakeOwner" tweet={fakeTweet} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders without owner correctly', () => {
    const component = renderer.create(<TweetCard tweet={fakeTweet} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('send click event when clicking avatar', async () => {
    const mockOnClick = jest.fn();
    render(
      <TweetCard
        owner="fakeOwner"
        tweet={fakeTweet}
        onAvatarClick={mockOnClick}
      />
    );

    const button = screen.getByTestId('avatar');
    userEvent.click(button);
    await waitFor(() => {
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
  });

  it('send click event when clicking delete', async () => {
    const mockOnClick = jest.fn();
    render(
      <TweetCard owner="fakeOwner" tweet={fakeTweet} onDelete={mockOnClick} />
    );

    const button = screen.getByTestId('delete');
    userEvent.click(button);
    await waitFor(() => {
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
  });
});
