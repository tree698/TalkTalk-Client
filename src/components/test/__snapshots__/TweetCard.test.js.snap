// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetCard renders with owner correctly 1`] = `
<li
  className="flex items-start justify-end mb-1 md:mb-2 lg:mb-3 text-[10px] sm:text-xs md:text-sm"
>
  <button
    className="order-1"
    data-testid="avatar"
    onClick={[Function]}
  >
    <div>
      <img
        alt="test"
        className="w-6 lg:w-8 h-6 lg:h-8 rounded-full ml-1"
        src="test"
      />
    </div>
  </button>
  <div
    className="w-3/5 p-2 rounded-tl-xl rounded-b-xl bg-test text-white bg-tweetOwner"
  >
    <div
      className="flex justify-between text-[9px]"
    >
      <p
        className="order-1"
      >
        test
      </p>
      <p>
        just now
      </p>
    </div>
    <div
      className="flex justify-between"
    >
      <p
        className="font-semibold"
      >
        test
      </p>
      <button
        className="hover:scale-105 hover:rotate-17 hover:text-black transition-all delay-150 duration-300 ease-in-out"
        data-testid="delete"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>
</li>
`;

exports[`TweetCard renders without owner correctly 1`] = `
<li
  className="flex items-start justify-start mb-1 md:mb-2 lg:mb-3 text-[10px] sm:text-xs md:text-sm"
>
  <button
    className=""
    data-testid="avatar"
    onClick={[Function]}
  >
    <div>
      <img
        alt="test"
        className="w-6 lg:w-8 h-6 lg:h-8 rounded-full mr-1"
        src="test"
      />
    </div>
  </button>
  <div
    className="w-3/5 p-2 rounded-tr-xl rounded-b-xl bg-tweetParticipant"
  >
    <div
      className="flex justify-between text-[9px]"
    >
      <p
        className=""
      >
        test
      </p>
      <p>
        just now
      </p>
    </div>
    <div
      className="flex justify-between"
    >
      <p
        className="font-semibold"
      >
        test
      </p>
    </div>
  </div>
</li>
`;
